client/tests/phpunit/Pact/Listener/PactTestListener.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace AppBundle\Pact\Listener;
+
+use PhpPact\Http\GuzzleClient;
+use PhpPact\Standalone\Exception\MissingEnvVariableException;
+use PhpPact\Standalone\MockService\MockServerConfigInterface;
+use PhpPact\Standalone\MockService\MockServerEnvConfig;
+use PhpPact\Standalone\MockService\Service\MockServerHttpService;
+use PHPUnit\Framework\AssertionFailedError;
+use PHPUnit\Framework\Test;
+use PHPUnit\Framework\TestListener;
+use PHPUnit\Framework\TestListenerDefaultImplementation;
+use PHPUnit\Framework\TestSuite;
+
+/**
+ * PACT listener that can be used with environment variables and easily attached to PHPUnit configuration.
+ * Class PactTestListener
+ */
+class PactTestListener implements TestListener
+{
+    use TestListenerDefaultImplementation;
+
+    /**
+     * Name of the test suite configured in your phpunit config.
+     *
+     * @var array
+     */
+    private $testSuiteNames;
+
+    /** @var MockServerConfigInterface */
+    private $mockServerConfig;
+
+    /** @var bool */
+    private $failed;
+
+    /**
+     * PactTestListener constructor.
+     *
+     * @param string[] $testSuiteNames test suite names that need evaluated with the listener
+     *
+     * @throws MissingEnvVariableException
+     */
+    public function __construct(array $testSuiteNames)
+    {
+        $this->testSuiteNames   = $testSuiteNames;
+        $this->mockServerConfig = new MockServerEnvConfig();
+    }
+
+    /**
+     * @param TestSuite $suite
+     *
+     * @throws \Exception
+     */
+    public function startTestSuite(TestSuite $suite): void
+    {
+        $httpService = new MockServerHttpService(new GuzzleClient(), $this->mockServerConfig);
+        $httpService->deleteAllInteractions();
+    }
+
+    public function addError(Test $test, \Throwable $t, float $time): void
+    {
+        $this->failed = true;
```

Comment:
Not sure of the internal working of pact or outputs, but is there any way to provide a bit more context around why the test error-ed? Or will this be part of the failing test?

Reply:
This is just a PHPUnit listener that's called if the test errors or fails. The normal PHPUnit output/exit code will happen as usual, but this allows us to stop generating a broken pact file (in `endTestSuite` below).

If the test fails because of something to do with Pact, details will be output like other tests: either with an error to stderr or with a "Expected value 1234 does not equal 4567"